test/functional_tests/test_12_submodule_update.py

```diff
@@ -190,6 +192,228 @@ def test_include(gitcache_ifc: GitcacheIfc):
     assert gitcache_ifc.db_field("clones", repo_sub2) is None
 
 
+@pytest.mark.parametrize(
+    "repo_url,submodule_url,final_submodule_url",
+    [
+        # Base URL "https://github.com/seeraven/submodule-example"
+        (
+            "https://github.com/seeraven/submodule-example",
+            "https://github.com/seeraven/dmdcache",
+            "https://github.com/seeraven/dmdcache",
+        ),
+        (
+            "https://github.com/seeraven/submodule-example",
+            "../dmdcache",
+            "https://github.com/seeraven/dmdcache",
+        ),
+        (
+            "https://github.com/seeraven/submodule-example",
+            "../something/../dmdcache",
+            "https://github.com/seeraven/dmdcache",
```

Comment:
I think for cases like this it worth making the behavior closer to what would git do, i.e. the submodule url should match the one as if generated by git itself: `https://github.com/seeraven/something/../dmdcache`.

And not to break the behavior of local mirror for gitcache, the `https://github.com/seeraven/something/../dmdcache` best to be translated into something like `github.com/seeraven/something/__/dmdcache`.

@seeraven what do you say?